tracer/src/Datadog.Trace/Util/HexString.cs

```diff
@@ -86,6 +125,33 @@ public static string ToHexString(ulong value, bool lowerCase = true)
         return ToHexString(bytes, lowerCase);
     }
 
+    /// <summary>
+    /// Converts the specified <see cref="TraceId"/> value into a hexadecimal string.
+    /// </summary>
+    [Pure]
+    public static string ToHexString(TraceId value, bool pad64To128 = true, bool lowerCase = true)
+    {
+        if (!pad64To128 && value.Upper == 0)
+        {
+            return ToHexString(value.Lower);
+        }
+
+        var upper = ReverseIfLittleEndian(value.Upper);
+        var lower = ReverseIfLittleEndian(value.Lower);
+
+#if NETCOREAPP3_1_OR_GREATER
+        Span<byte> bytes = stackalloc byte[16];
```

Comment:
add comment about not using `MemoryMarshal.Write(bytes, ref value)` because it tries to handle endianness